From 646ba3d4e1ab8bd7b1a725b68815bdb4d7081a9d Mon Sep 17 00:00:00 2001
From: kvnp <kevin.pfeifer8@gmail.com>
Date: Sun, 8 Nov 2020 13:12:00 +0100
Subject: [PATCH] fix web support

---
 SlidingUpPanel.js | 27 ++++++++++++++++++++++++++-
 1 file changed, 26 insertions(+), 1 deletion(-)

diff --git a/SlidingUpPanel.js b/SlidingUpPanel.js
index a5553dc..2166bcf 100644
--- a/SlidingUpPanel.js
+++ b/SlidingUpPanel.js
@@ -34,7 +34,8 @@ const keyboardHideEvent = Platform.select({
 const usableHeight = visibleHeight() - statusBarHeight()
 
 class SlidingUpPanel extends React.PureComponent {
-  static propTypes = {
+  static propTypes = Platform.OS != "web"
+  ? {
     height: PropTypes.number,
     animatedValue: PropTypes.instanceOf(Animated.Value),
     draggableRange: PropTypes.shape({
@@ -60,6 +61,30 @@ class SlidingUpPanel extends React.PureComponent {
     backdropStyle: ViewPropTypes.style,
     children: PropTypes.oneOfType([PropTypes.element, PropTypes.func])
   }
+  : {
+    height: PropTypes.number,
+    animatedValue: PropTypes.instanceOf(Animated.Value),
+    draggableRange: PropTypes.shape({
+      top: PropTypes.number,
+      bottom: PropTypes.number
+    }),
+    snappingPoints: PropTypes.arrayOf(PropTypes.number),
+    minimumVelocityThreshold: PropTypes.number,
+    minimumDistanceThreshold: PropTypes.number,
+    avoidKeyboard: PropTypes.bool,
+    onBackButtonPress: PropTypes.func,
+    onDragStart: PropTypes.func,
+    onDragEnd: PropTypes.func,
+    onMomentumDragStart: PropTypes.func,
+    onMomentumDragEnd: PropTypes.func,
+    onBottomReached: PropTypes.func,
+    allowMomentum: PropTypes.bool,
+    allowDragging: PropTypes.bool,
+    showBackdrop: PropTypes.bool,
+    backdropOpacity: PropTypes.number,
+    friction: PropTypes.number,
+    children: PropTypes.oneOfType([PropTypes.element, PropTypes.func])
+  }
 
   static defaultProps = {
     height: usableHeight,
-- 
2.29.2.windows.2